internal/api/handlers.go

```diff
@@ -252,6 +257,105 @@ func DeleteCertificate(env *Environment) http.HandlerFunc {
 	}
 }
 
+// GetUserAccounts returns all users from the database
+func GetUserAccounts(env *Environment) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		users, err := env.DB.RetrieveAllUsers()
+		if err != nil {
+			logErrorAndWriteResponse(err.Error(), http.StatusInternalServerError, w)
+			return
+		}
+		apiUsers := make([]certdb.User, 0, len(users))
+		for _, user := range users {
+			apiUsers = append(apiUsers, certdb.User{
+				ID:          user.ID,
+				Username:    user.Username,
+				Permissions: user.Permissions,
+			})
+		}
+		body, err := json.Marshal(apiUsers)
+		if err != nil {
+			logErrorAndWriteResponse(err.Error(), http.StatusInternalServerError, w)
+			return
+		}
+		if _, err := w.Write(body); err != nil {
+			logErrorAndWriteResponse(err.Error(), http.StatusInternalServerError, w)
+		}
+	}
+}
+
+// GetUserAccount receives an id as a path parameter, and
+// returns the corresponding User Account
+func GetUserAccount(env *Environment) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := r.PathValue("id")
+		userAccount, err := env.DB.RetrieveUser(id)
+		if err != nil {
+			if err.Error() == "user id not found" {
+				logErrorAndWriteResponse(err.Error(), http.StatusNotFound, w)
+				return
+			}
+			logErrorAndWriteResponse(err.Error(), http.StatusInternalServerError, w)
+			return
+		}
+		apiUser := certdb.User{
+			ID:          userAccount.ID,
+			Username:    userAccount.Username,
+			Permissions: userAccount.Permissions,
+		}
+
+		body, err := json.Marshal(apiUser)
```

Comment:
This is why we modified the json tags on the structs. Read below on why we set the password to `""`
```suggestion
		userAccount.Password = ""
		body, err := json.Marshal(userAccount)
```

Reply:
✅